spec/abci/abci++_basic_concepts.md

```diff
@@ -233,9 +239,12 @@ from block execution (`FinalizeBlock` calls), and not through
 any other kind of request. This is the only way to ensure all nodes see the same
 transactions and compute the same results.
 
-Some Applications may choose to implement immediate execution, which entails executing the blocks
+Some Applications may choose to execute blocks asynchronously to optimistically speed
```

Comment:
Did you mean synchronously?
If you execute blocks asynchronously, then you can't reject the invalid blocks.
Or maybe I'm not fully understanding

Reply:
Remember, what we achieve with immediate execution is the 100% guarantee that no TX included in a block will fail.
The app may choose to execute the block in parallel by replying `ACCEPT` right away, but then it cannot reject blocks based on its execution (in the current version of ABCI).

Reply:
I really meant asynchronouusly, by replying ACCEPT right away. I modified the previous change to explicitly call out immediate execution (maybe a better name would be early execution) and the following change to discuss both immediate execution and optimistic/asynchronous/parallel execution.

Reply:
Ok I'll re-read in the next pass